From 89d7a5a417dccbdc97107d30a6678e94ced9ca5d Mon Sep 17 00:00:00 2001
From: Felix Geyer <debfx@ubuntu.com>
Date: Wed, 10 Sep 2014 08:55:00 -0400
Subject: [PATCH] Fix CMAKE_HOST_SYSTEM_PROCESSOR detection on GNU/Hurd
 (#15147)

On Debian GNU/Hurd 'uname -p' prints "unknown", so use 'uname -m'
instead.
---
 Modules/CMakeDetermineSystem.cmake |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Modules/CMakeDetermineSystem.cmake b/Modules/CMakeDetermineSystem.cmake
index 1c0941a..fe292ea 100644
--- a/Modules/CMakeDetermineSystem.cmake
+++ b/Modules/CMakeDetermineSystem.cmake
@@ -47,7 +47,7 @@ if(CMAKE_HOST_UNIX)
   if(CMAKE_UNAME)
     exec_program(uname ARGS -s OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_NAME)
     exec_program(uname ARGS -r OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_VERSION)
-    if(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux|CYGWIN.*|Darwin")
+    if(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux|CYGWIN.*|Darwin|^GNU$")
       exec_program(uname ARGS -m OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
         RETURN_VALUE val)
       if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin" AND
-- 
1.7.10.4

